[Navi21] test_activation quietly fails

```
35/92 Test  #1: test_activation .......................................***Failed  394.89 sec
/dockerx/github/MIOpen/build/release.hip-clang.nodev/bin/test_activation --float --input 1, 1, 1, 1 --alpha 0.95 --beta 2.3 --gamma 3.4 --mode ELU --packed 1.
error: 0
Max diff: 0
Forward Activation: miopenActivationELU
Input tensor: 1, 1, 1, 1...
/dockerx/github/MIOpen/build/release.hip-clang.nodev/bin/test_activation --float --input 1, 16, 4096, 4096 --alpha 0.95 --beta 2.3 --gamma 3.4 --mode ELU --packed 1.
error: 0
Max diff: 0
Forward Activation: miopenActivationELU
Input tensor: 1, 16, 4096, 4096
/dockerx/github/MIOpen/build/release.hip-clang.nodev/bin/test_activation --float --input 1, 16, 4096, 4096 --alpha 0.95 --beta 2.3 --gamma 3.4 --mode ELU --packed 1.
CMake Error at test_test_activation.cmake:7 (message):
  Test failed
```
but no other signs of error.

_Originally posted by @atamazov in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1333#issuecomment-989256733_

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -179,9 +179,6 @@ option( WORKAROUND_ISSUE_1187 "" ${WORKAROUND_ISSUE_1187_DEFAULT})
 set_var_to_condition(WORKAROUND_ISSUE_1148_DEFAULT MIOPEN_TEST_GFX103X AND MIOPEN_TEST_FLOAT)
 option( WORKAROUND_ISSUE_1148 "" ${WORKAROUND_ISSUE_1148_DEFAULT})
 
-set_var_to_condition(WORKAROUND_ISSUE_1334_DEFAULT MIOPEN_TEST_GFX103X AND MIOPEN_TEST_FLOAT)
-option( WORKAROUND_ISSUE_1334 "" ${WORKAROUND_ISSUE_1334_DEFAULT})
-
 if(MIOPEN_TEST_HALF)
     if(MIOPEN_BACKEND_OPENCL)
 	    set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm)
@@ -316,10 +313,6 @@ function(add_test_executable TEST_NAME)
     set_tests_properties(${TEST_NAME} PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED")
     if(WORKAROUND_ISSUE_1148
         AND (${TEST_NAME} MATCHES "test_soft_max") )
-        set_tests_properties(${TEST_NAME} PROPERTIES RUN_SERIAL On)
-    endif()
-    if(WORKAROUND_ISSUE_1334
-        AND (${TEST_NAME} MATCHES "test_activation") )
         set_tests_properties(${TEST_NAME} PROPERTIES RUN_SERIAL On)
     endif()
     if(NOT MIOPEN_EMBED_DB STREQUAL "")

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -364,6 +364,7 @@ set(LONG_TESTS
     test_pooling2d
     test_conv_igemm_mlir
     test_conv_igemm_mlir_xdlops
+    test_activation
     )
 
 function(rate_added_test NAME)

--- a/test/activation.cpp
+++ b/test/activation.cpp
@@ -38,6 +38,7 @@
 #include "driver.hpp"
 #include "get_handle.hpp"
 #include "tensor_holder.hpp"
+#include "random.hpp"
 #include "verify.hpp"
 
 std::string to_name(miopenActivationMode_t m)
@@ -167,6 +168,18 @@ struct select_first
 {
     template <class T>
     auto operator()(const T& x) MIOPEN_RETURNS(x.first); // NOLINT (readability-const-return-type)
+};
+
+// Borrowed from conv_common.h
+struct scalar_gen_random_float
+{
+    double min_val = 0;
+    double max_val = 1;
+
+    double operator()() const
+    {
+        return min_val + (max_val - min_val) * double(GET_RAND()) / RAND_MAX;
+    }
 };
 
 template <class T>
@@ -304,6 +317,12 @@ struct activation_driver : test_driver
     void run(miopenActivationMode_t m, Forward f, Backward b)
     {
         auto desc = make_descriptor(m);
+
+        auto gen_sign_value = [=](auto... is) {
+            return scalar_gen_random_float{-4, 4}() * tensor_elem_gen_checkboard_sign{}(is...);
+        };
+        input.generate(gen_sign_value);
+
         auto out  = verify(verify_forward_activation<T>{input, desc}, f);
         auto dout = out.first;
         dout.generate([&](int n, int c, int h, int w) {

